src/proxy.js

```diff
@@ -297,3 +402,16 @@ function makeIterable(next) {
         next
     })
 }
+
+function makeTrapsForGetters(getters) {
+    return {
+        get(state, prop, receiver) {
+            return getters.hasOwnProperty(prop)
+                ? getters[prop](state, prop, receiver)
+                : Reflect.get(state, prop, receiver)
+        },
+        ownKeys(state) {
+            return Reflect.ownKeys(source(state))
```

Comment:
This trap can be removed

Reply:
Wouldn't it return keys of our state then?

Reply:
Good catch. 👍  I remember adding this trap to fix an internal issue, then removing it once realizing it's not necessary to fix the internals. I didn't think of users calling `Object.keys` on their Set/Map objects, which definitely shouldn't return the keys of the draft-state.

Reply:
If we care about the `Object.keys` edge case, we should also support `Object.getOwnPropertyDescriptor` by adding a `getOwnPropertyDescriptor` trap.

Reply:
And what about the `getPrototypeOf` or `setPrototypeOf` traps? And the `defineProperty`, `set`, `deleteProperty` traps? It gets to be a bit much, but for the sake of correctness, we should probably do it?

Reply:
I guess we're gonna need tests to cover them as well.